### ISSUE TITLE
Incorrect IDE0047 Parentheses can be removed diagnostic

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#52589]
**Version Used**: 4e4c823

[SharpLab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4AEBMBGAsAKBQZgAJ1CBhQgb10JuKJQBZCBZACgHtgArOAYxkIAPAJSVqtCQEsAZoVaDCkqHID6w4eIk0qOLbQC+mwoZz6gA===)

```cs
using System;
public class C {
    public void M(object x) {
        if (x is (_)) // info IDE0047: Parentheses can be removed
        {
        }
    }
}
```

**Expected Behavior**: No IDE0047 diagnostic

**Actual Behavior**: IDE0047 is given, but when the user applies the fix, it introduces a compile error `error CS0246: The type or namespace name '_' could not be found` because the expression is now considered to be an "is type" expression instead of "is pattern" and old binding rules apply.

I think this scenario may be changed if @jmarolf's suggested feature to remove non-lexical contextual keywords is implemented.

**Comment by RikkiGibson** [comment]
Related to #50025, except the ambiguity is between a discard and the type `_` instead of between a constant and a type.
